.github/workflows/submitty_ci.yml

```diff
@@ -300,195 +299,48 @@ jobs:
-              sudo rm -rf /etc/apache2/sites-enabled/*
-              sudo cp -f .setup/apache/submitty.conf /etc/apache2/sites-available/submitty.conf
-              sudo sed -e "s/Require host __your_domain__/Require all granted/g" --in-place /etc/apache2/sites-available/submitty.conf
-              sudo cp .setup/apache/www-data /etc/apache2/suexec/www-data
-              sudo chmod 0640 /etc/apache2/suexec/www-data
-              sudo a2ensite submitty
-              sudo bash -c 'echo "export PATH=$PATH" >> /etc/apache2/envvars'
-              sudo apache2ctl -t
-              sudo service php${PHP_VER}-fpm restart
-              sudo service apache2 restart
-
-          - name: Set up nginx
-            run: |
-              cd $SUBMITTY_REPOSITORY
-              sudo mkdir /etc/systemd/system/nginx.service.d
-              sudo printf "[Service]\nExecStartPost=/bin/sleep 0.1\n" | sudo tee /etc/systemd/system/nginx.service.d/override.conf
-              sudo systemctl daemon-reload
-              sudo rm -rf /etc/nginx/sites-available/*
-              sudo rm -rf /etc/nginx/sites-enabled/*
-              sudo cp -f .setup/nginx/submitty.conf /etc/nginx/sites-available/submitty.conf
-              sudo chmod 644 /etc/nginx/sites-available/submitty.conf
-              sudo ln -s /etc/nginx/sites-available/submitty.conf /etc/nginx/sites-enabled/submitty.conf
-              sudo service nginx restart
-
-          - name: Run git tests
-            run: |
-              cd $SUBMITTY_REPOSITORY
-              SEMESTER=$(python3 -c 'from datetime import datetime; today = datetime.today(); semester = ("s" if today.month < 7 else "f") + str(today.year)[-2:]; print(semester)')
-              sudo python3 /usr/local/submitty/bin/generate_repos.py ${SEMESTER} sample open_homework
-              bash tests/git_test.sh
-
-          - uses: nanasess/setup-chromedriver@master
-          - name: Setup chromedriver
-            run: |
-              sudo systemctl restart submitty_autograding_worker
-              sudo systemctl restart submitty_autograding_shipper
-              export DISPLAY=:99
-              chromedriver --url-base=/wd/hub &
-
+          - name: Checkout repository
+            uses: actions/checkout@v3
+          - uses: ./.github/actions/set-up-composite
           - name: Run selenium e2e tests with pam auth
             run: |
               pushd ${SUBMITTY_REPOSITORY}
               echo "Authentication Method => $(sudo jq -r ".authentication_method" /usr/local/submitty/config/database.json)"
               TEST_URL="http://localhost" python3 -m unittest discover -v --start-directory tests
               sudo bash tests/test_site_error_log.sh
 
-          # TODO: Remove this block after upgrading jsPDF
+    e2eCypressOne:
+        needs:
+          - js-lint
+          - js-unit
+          - php-lint
+          - php-unit
+          - python-lint
+          - python-unit
+          - css-lint
+          - shellcheck
```

Comment:
We can shave off another 3-4 minutes from the overall running time by not adding these as dependencies.

Reply:
Something like [this](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast) might work.  If I'm understanding the docs correctly, it should kill all the other running jobs whenever a job with the fail fast option fails.